src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -1681,9 +1681,26 @@ internal bool ShutdownLoop
                 }
             }
 
+            /// <summary>
+            /// Call WaitForCommEvent (which is a thread function for a background thread)
+            /// within an exception handler, so that unhandled exceptions in WaitForCommEvent
+            /// don't cause process termination
+            /// </summary>
+            internal void SafelyWaitForCommEvent()
+            {
+                try
+                {
+                    WaitForCommEvent();
+                }
+                catch(Exception ex)
```

Comment:
@willdean what I was suggesting is to put in a Debug.Fail just for the exception types we don't expect. If we know that ObjectDisposedException may happen, then presumably Debug.Fail would have no purpose if it fired then.?